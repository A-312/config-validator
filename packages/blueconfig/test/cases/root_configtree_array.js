const blueconfig = require('blueconfig')
const LISTOFERRORS = require('blueconfig/lib/error.js').LISTOFERRORS

exports.formats = {
  'Object[]': {
    validate: function(children, schema, fullname) {
      const errors = []

      if (!Array.isArray(children)) {
        throw new Error('must be an Array')
      }

      children.forEach((child, keyname) => {
        try {
          const conf = blueconfig(schema.children).merge(children[keyname]).validate()
          this.set(fullname + '.' + keyname, conf.getProperties())
        } catch (err) {
          err.parent = fullname + '.' + keyname
          errors.push(err)
        }
      })

      if (errors.length !== 0) {
        throw new LISTOFERRORS(errors)
      }
    }
  }
}

exports.conf = {
  format: 'Object[]',
  default: [],
  children: {
    // conf[].xxx:
    name: {
      format: 'String',
      default: undefined,
      required: true
    },
    population: {
      format: 'int',
      default: 0
    },
    subregion: {
      format: 'String',
      default: 'Europe'
    }
  }
}

exports.dataType = 'data'
exports.data = [{
  name: 1,
  population: '83783942 persons',
  subregion: 'Western Europe'
},
{
  name: 'France',
  population: 65273511
},
{
  name: 'Italy',
  population: 60461826,
  subregion: 2
}]
