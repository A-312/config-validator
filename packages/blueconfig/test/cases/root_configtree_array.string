[{
    "name": 1,
    "population": "83783942 persons",
    "subregion": "Western Europe"
  },
  {
    "name": "France",
    "population": 65273511,
    "subregion": "Europe"
  },
  {
    "name": "Italy",
    "population": 60461826,
    "subregion": 2
}]