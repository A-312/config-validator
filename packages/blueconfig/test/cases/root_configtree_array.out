Validate failed because wrong value(s):
  - root: Custom format "Object[]" tried to validate something and failed:
    1) 0:
      - name: must be of type String: value was 1
    2) 2:
      - subregion: must be of type String: value was 2, getter was `value`